pkg/pgclient/config.go

```diff
@@ -46,13 +48,17 @@ func ParseFlags(cfg *Config) *Config {
 	flag.Uint64Var(&cfg.MetricsCacheSize, "metrics-cache-size", pgmodel.DefaultMetricCacheSize, "Maximum number of metric names to cache.")
 	flag.IntVar(&cfg.WriteConnectionsPerProc, "db-writer-connection-concurrency", 4, "Maximum number of database connections for writing per go process.")
 	flag.IntVar(&cfg.MaxConnections, "db-connections-max", -1, "Maximum number of connections to the database that should be opened at once. It defaults to 80% of the maximum connections that the database can handle.")
+	flag.StringVar(&cfg.ServiceUrl, "service-url", DefaultServiceUrlFromDefaultFlags, "TimescaleDB/Vanilla Postgres service URL.")
 	return cfg
 }
 
 // GetConnectionStr returns a Postgres connection string
 func (cfg *Config) GetConnectionStr() string {
-	return fmt.Sprintf("host=%v port=%v user=%v dbname=%v password='%v' sslmode=%v connect_timeout=10",
-		cfg.Host, cfg.Port, cfg.User, cfg.Database, cfg.Password, cfg.SslMode)
+	if cfg.ServiceUrl == DefaultServiceUrlFromDefaultFlags {
+		return fmt.Sprintf("host=%v port=%v user=%v dbname=%v password='%v' sslmode=%v connect_timeout=10",
+			cfg.Host, cfg.Port, cfg.User, cfg.Database, cfg.Password, cfg.SslMode)
```

Comment:
We should validate that we are only setting either the URI or these parameters but not both. We should throw an error if setting a combination. The reason is that otherwise users may think we'll combine the URI with these in some way.